src/lib/server/xlat_func.c

```diff
@@ -1711,6 +1711,8 @@ static ssize_t xlat_func_explode(TALLOC_CTX *ctx, char **out, size_t outlen,
 
 	if (*p == '\0') goto arg_error;
 
+	if (strlen(p) > 1) goto arg_error;
```

Comment:
perhaps `if (p[1]) goto arg_error`.  You don't care about the length of the string.  You care that there's *anything* after the first character.

Reply:
done